src/fragments/lib/auth/common/mfa/flows.mdx

```diff
@@ -0,0 +1,206 @@
+
+`continueSignInWithTOTPSetup` step signifies that the user must set up TOTP before they can sign in. The step returns an associated value of type `TOTPSetupDetails` which would be used for generating TOTP. `TOTPSetupDetails` provides a helper method called `getSetupURI` that can be used to generate a URI, which can be used by native password managers for TOTP association. For example. if the URI is used on Apple platforms, it will trigger the platform's native password manager to associate TOTP with the account. For more advanced use cases, `TOTPSetupDetails` also contains the `sharedSecret` that will be used to either generate a QR code or can be manually entered into an Authenticator app. 
+ 
+Once the authenticator app is set up, the user can generate a TOTP code and provide it to the library to complete the sign in process.
+
+import iosSignIn from "/src/fragments/lib/auth/ios/totp/sign_in.mdx";
+
+<Fragments fragments={{ios: iosSignIn}} />
+
+The TOTP code can be obtained from the user via a text field or any other means. Once the user provides the TOTP code, call `confirmSignIn` with the TOTP code as the `challengeResponse` parameter.
+
+import iosConfirmSignIn from "/src/fragments/lib/auth/ios/totp/confirm_sign_in.mdx";
+
+<Fragments fragments={{ios: iosConfirmSignIn}} />
+
+### Enabling TOTP after a user is signed in
+
+TOTP MFA can be set up after a user has signed in. This can be done when the following conditions are met:
+
+- MFA is marked as **Optional** or **Required** in Cognito User Pool.
+- TOTP is marked as an enabled MFA method in Cognito user pool
+
+TOTP setup can be achieved by calling `setUpTOTP` and `verifyTOTPSetup` APIs that are available in the `Auth` category.
+
+Invoke the `setUpTOTP` API to generate a `TOTPSetupDetails` object that contains the `sharedSecret` which will be used to either to generate a QR code or can be manually entered into an Authenticator app.
+
+**`setUpTOTP` API**
+
+import iosSetUpTOTP from "/src/fragments/lib/auth/ios/totp/set_up_totp.mdx";
+
+<Fragments fragments={{ios: iosSetUpTOTP}} />
+
+Once the authenticator app is set up, the user can get a TOTP code that is generated by the Authenticator app and can provide it to the library to using the `verifyTOTPSetup` API to complete the TOTP setup process.
+
+**`verifyTOTPSetup` API**
+
+import iosVerifyTOTPSetup from "/src/fragments/lib/auth/ios/totp/verify_totp_setup.mdx";
+
+<Fragments fragments={{ios: iosVerifyTOTPSetup}} />
+
+### Recovering from a lost TOTP device
+
+If a user loses access to their TOTP device, they would need to contact an administrator to help get access to their account. Based on the Cognito User Pool configuration, the administrator can use the [AdminSetUserMFAPreference](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminSetUserMFAPreference.html) to either change the MFA preference to a different MFA method or to disable MFA for the user.
```

Comment:
I would think this would want to be called out in the warning. This is a huge deal IMO.

Reply:
Done.